File: Analysis of algorithms Exam/insertion.java
import java.util.Scanner;

class insertion{
    static void readArray(int a[],int n){
        Scanner sc=new Scanner(System.in);
        for(int i=0;i<n;i++){
            a[i]=sc.nextInt();
        }
    }
    static void displayArray(int a[],int n){
        System.out.println("Array details");
        System.out.println();
        System.out.print("{"+"\t");
         for(int i=0;i<n;i++){
             System.out.print(a[i]+"\t");
         }
         System.out.print("}");
         System.out.println();
     }
    static void insertionSort(int a[],int n){
        for(int j=1;j<n;j++){
            int i=j-1;
            int key=a[j];
            while(i>=0 && a[i]>key){
                a[i+1]=a[i];
                i--;
            }
            a[i+1]=key;
            System.out.println("__________________________________________________________________________________");
            System.out.println();
            System.out.println("after pass "+(j));
            displayArray(a, n);
            System.out.println();
        }

    }

    public static void main(String args[]){
        Scanner sc=new Scanner(System.in);
        System.out.println("enter the size of the array:");
        int n=sc.nextInt();
        int a[]=new int[n];
        System.out.println("__________________________________________________________________________________");
        System.out.println("enter the array details");
        readArray(a, n);
        System.out.println("__________________________________________________________________________________");
        displayArray(a, n);
        System.out.println("__________________________________________________________________________________");
        insertionSort(a, n);
        System.out.println("__________________________________________________________________________________");
        displayArray(a, n);
    }
}

File: insertion.c
#include<stdio.h>
#include<conio.h>
void readArray(int n,int array[n])
{
    printf("\nEnter the array elements :\n");
    for(int i=0;i<n;i++)
    {
     scanf("%d",&array[i]);
    }
    
}
void displayArray(int n,int array[n])
{
    printf("\nThe array elements are as follows:\n");
    for(int i=0;i<n;i++)
    {
    printf("%d \n",array[i]); 
    }
    
}
void insertionSort(int n, int array[n])
{
for(int j=1;j<n;j++)
{
    int key=array[j];
    int i=j-1;
    while((i>=0)&&(array[i]>key))
    {
        array[i+1]=array[i];
        i=i-1;
    }
    array[i+1]=key;
        printf("after iteration %d",j);
        displayArray(n,array);
}
}
void main()
{
   int n;
   printf("enter the array size:");
   scanf("%d",&n);
   int array[n];
   readArray(n,array);
   printf("\nbefore swapping the array elements:\n");
   displayArray(n,array);
   insertionSort(n,array);
   displayArray(n,array);
}